test/built-ins/TypedArray/prototype/toString/this-is-not-object.js

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.toString
+description: Throws a TypeError exception when `this` is not Object
+info: >
+  22.2.3.29 %TypedArray%.prototype.toString ( )
+
+  ...
+
+  22.2.3.15 %TypedArray%.prototype.join ( separator )
+
+  This function is not generic. ValidateTypedArray is applied to the this value
+  prior to evaluating the algorithm. If its result is an abrupt completion that
+  exception is thrown instead of evaluating the algorithm.
+
+  22.2.3.5.1 Runtime Semantics: ValidateTypedArray ( O )
+
+  1. If Type(O) is not Object, throw a TypeError exception.
```

Comment:
@leobalter, what do you think about simply checking for `TypedArray.prototype.toString === Array.prototype.toString` instead? According to the spec, they should have the same value.